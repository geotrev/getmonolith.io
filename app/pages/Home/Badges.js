import React from "react"
import Markdown from "react-markdown"
import statusBadges from "./markdown/badges.md"

const Badges = () => <Markdown source={statusBadges} escapeHtml={false} />

export default Badges
