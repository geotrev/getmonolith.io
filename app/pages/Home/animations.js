export const ANIMATION_DATA = [
  {
    title: "Tiny",
    subtitle:
      "CSS and JS under 14kb minified + gzipped; you can be assured performance isnâ€™t an issue.",
    animation: null,
  },
  {
    title: "Modular",
    subtitle:
      "Remove and namespace pieces of the framework to prevent collisions and take only what you need.",
    animation: null,
  },
  {
    title: "Configurable",
    subtitle:
      "Built for a great developer experience, you can customize and extend the library with ease.",
    animation: null,
  },
  {
    title: "Accessible",
    subtitle:
      "Interactive components are designed with WAI-ARIA guidelines in mind to ensure your HTML is accessible.",
    animation: null,
  },
]
