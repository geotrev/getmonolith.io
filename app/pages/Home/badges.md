[![CircleCI](https://circleci.com/gh/geotrev/undernet/tree/master.svg?style=svg)](https://circleci.com/gh/geotrev/undernet/tree/master) [![devDependencies Status](https://david-dm.org/geotrev/undernet/dev-status.svg)](https://david-dm.org/geotrev/undernet?type=dev) [![npm version](https://badge.fury.io/js/undernet.svg)](https://badge.fury.io/js/undernet)
