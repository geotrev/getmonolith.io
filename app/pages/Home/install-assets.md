```html
<html>
  <head>
    …
    <link rel=“stylesheet” href=“path/to/undernet.min.css />
  </head>
  <body>
    …
    <script type=“text/javascript” src=“path/to/undernet.bundle.min.js”></script>
    <script>
      document.addEventListener(“DOMContentLoaded”, Undernet.start())
    </script>
  </body>
</html>
```
