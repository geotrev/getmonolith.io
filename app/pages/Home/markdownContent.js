import React from "react"
import Markdown from "react-markdown"
import Loadable from "react-loadable"

import installNpm from "./installNpm.md"
import installAssets from "./installAssets.md"

const loader = () => import("./Badges")
const loading = () => <span className="is-visually-hidden">Loading badges</span>
export const StatusBadges = Loadable({ loader, loading })

export const InstallNpm = () => <Markdown source={installNpm} escapeHtml={false} />
export const InstallAssets = () => <Markdown source={installAssets} escapeHtml={false} />
