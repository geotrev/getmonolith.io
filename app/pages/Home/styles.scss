@import "app/global";

@mixin bg-gradient {
  background: #3f48af;
  background: -webkit-linear-gradient(right, #3e46af, #d15ded);
  background: -moz-linear-gradient(right, #3e46af, #d15ded);
  background: linear-gradient(to left, #3e46af, #d15ded);
}

#home {
  width: 100%;

  .animations {
    background-color: $color-dark-blue;

    .animated-icon {
      width: 125px;
      margin: 0 auto;
    }

    p {
      margin: 0 8px;

      @include create-breakpoint-down(lg) {
        margin-bottom: 24px;
      }
    }
  }

  .hero {
    background-image: url("../../assets/images/grid-bg.svg");
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;

    h1 {
      margin-bottom: 0;
    }

    h2 {
      font-weight: normal;
    }

    a {
      margin: $global-space 8px $global-space 8px;
    }

    .un-version {
      color: $color-gray800;
    }

    .badges {
      display: inline-flex;
      margin: 0 auto;
      flex: 0 1 auto;
      padding: 5px 5px 0 5px;

      p {
        margin: 0;
        text-align: center;
      }

      a {
        margin: 0;
      }
    }
  }

  .has-gradient {
    @include bg-gradient;
    border-color: #3f48af;
    border-width: 2px;
    border-style: solid;
    opacity: 0.75;
    font-weight: bold;

    &:hover,
    &:focus,
    &:active {
      opacity: 1;
    }
  }
}
