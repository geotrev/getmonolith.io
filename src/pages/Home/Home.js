import React, { Component } from "react"
import "./Home.scss"
import { ScrollUpOnMount } from "helpers"

import { Link } from "react-router-dom"
import ChevronRight from "react-feather/dist/icons/chevron-right"
import Markdown from "react-markdown"

import statusBadges from "./badges.md"
import installNpm from "./install-npm.md"
import installAssets from "./install-assets.md"
import tinySvg from "assets/images/tiny.svg"
import modSvg from "assets/images/modular.svg"
import configSvg from "assets/images/configurable.svg"
import a11ySvg from "assets/images/accessible.svg"

export default class Home extends Component {
  render() {
    return (
      <div id="home">
        <ScrollUpOnMount />

        <div className="small-section fluid grid hero">
          <div className="row">
            <div className="xsmall-12 columns has-center-text">
              <h1 className="h3">Modular front-end development. No strings.</h1>
            </div>

            <div className="xsmall-12 columns has-center-text">
              <Link to="docs/getting-started" className="medium button has-feather">
                Download <ChevronRight size={20} />
              </Link>
              <Link to="docs/overview" className="primary medium button has-gradient has-feather">
                Learn More <ChevronRight size={20} />
              </Link>
            </div>

            <div className="xsmall-12 columns has-center-text">
              <p className="un-version has-no-margin">Version 2.1.0</p>
            </div>

            <div className="xsmall-12 columns badges">
              <Markdown source={statusBadges} escapeHtml={false} />
            </div>
          </div>
        </div>

        <div className="small-section fluid grid perks">
          <div className="row collapsed">
            <div className="wide grid">
              <ul className="row is-unstyled-list">
                <li className="large-3 small-6 xsmall-12 columns has-center-text">
                  <img className="home-icon" src={tinySvg} role="presentation" alt="tiny" />
                  <h2 className="h6 has-white-text">Tiny</h2>
                  <p className="has-white-text">
                    CSS and JS are only 8.5kb minified + gzipped; you can be assured performance
                    isnâ€™t an issue.
                  </p>
                </li>
                <li className="large-3 small-6 xsmall-12 columns has-center-text">
                  <img className="home-icon" src={modSvg} role="presentation" alt="modular" />
                  <h2 className="h6 has-white-text">Modular</h2>
                  <p className="has-white-text">
                    Include only the pieces you need, or even namespace the components for existing
                    projects.
                  </p>
                </li>
                <li className="large-3 small-6 xsmall-12 columns has-center-text">
                  <img
                    className="home-icon"
                    src={configSvg}
                    role="presentation"
                    alt="configurable"
                  />
                  <h2 className="h6 has-white-text">Configurable</h2>
                  <p className="has-white-text">
                    Built for a great developer experience, you can customize and extend the library
                    with ease.
                  </p>
                </li>
                <li className="large-3 small-6 xsmall-12 columns has-center-text">
                  <img className="home-icon" src={a11ySvg} role="presentation" alt="a11y" />
                  <h2 className="h6 has-white-text">Accessible</h2>
                  <p className="has-white-text">
                    Undernet is designed WAI-ARIA guidelines in mind to ensure your project is
                    accessible.
                  </p>
                </li>
              </ul>
            </div>
          </div>
        </div>

        <div className="small-section narrow grid">
          <div cassName="row">
            <div className="xsmall-12 columns has-center-text">
              <h2 className="h6">Painless Setup</h2>
              <p>Install with npm:</p>
              <Markdown source={installNpm} escapeHtml={false} />
              <br />
              <p>Or simply link to minified assets:</p>
              <Markdown source={installAssets} escapeHtml={false} />
              <br />
              <p>See how Undernet can improve your developer experience!</p>
              <Link to="docs/overview" className="primary medium button has-gradient has-feather">
                Learn More <ChevronRight size={20} />
              </Link>
            </div>
          </div>
        </div>
      </div>
    )
  }
}
