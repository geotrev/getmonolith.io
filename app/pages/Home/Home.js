import React, { useEffect } from "react"
import { Link } from "react-router-dom"
import Prism from "prismjs"
import lottie from "lottie-web"
import ChevronRight from "react-feather/dist/icons/chevron-right"

import ScrollUpOnMount from "app/components/ScrollUpOnMount"
import SetMeta from "app/components/SetMeta"
import PageHeader from "app/components/PageHeader"
import { downloadPath, introductionPath } from "app/routes"

import pkg from "projectRoot/package.json"
import { StatusBadges, InstallNpm, InstallAssets } from "./markdownContent"
import { ANIMATION_DATA } from "./animations"

import "./styles.scss"

export default function Home() {
  const loadAnimations = () => {
    ANIMATION_DATA.forEach(data => {
      const name = data.title.toLowerCase()

      const loadedAnimation = lottie.loadAnimation({
        container: document.getElementById(`animated-${name}`),
        renderer: "svg",
        loop: false,
        autoplay: false,
        path: `assets/${name}.json`,
        name,
      })

      data.animation = loadedAnimation
    })

    window.setTimeout(() => {
      ANIMATION_DATA.forEach(instance => instance.animation.play())
    }, 1500)
  }

  const componentUnmountFunction = () => {
    ANIMATION_DATA.forEach(instance => instance.animation.destroy())
  }

  const observedStateOnMount = []
  useEffect(() => {
    Prism.highlightAll()
    loadAnimations()

    return componentUnmountFunction
  }, observedStateOnMount)

  const renderAnimations = () => {
    return ANIMATION_DATA.map(animation => {
      const animationName = animation.title.toLowerCase()

      return (
        <li
          className="is-lg-3 is-sm-6 is-xs-12 column has-text-center has-no-p-block-end"
          key={animation.title}
        >
          <div className="animated-icon" id={`animated-${animationName}`} />
          <h2 className="h6 has-white-text-color">{animation.title}</h2>
          <p className="has-white-text-color">{animation.subtitle}</p>
        </li>
      )
    })
  }

  return (
    <div id="home">
      <ScrollUpOnMount />

      {/*
       * Title is set here _and_ in public/index.html...
       * Doing so prevents title changes on other pages from
       * persisting if a visitor returns to the home page.
       */}
      <SetMeta
        title="A modular, configuration-first front-end framework. No strings."
        description="Undernet is a highly customizable web framework for building websites."
      />

      <div className="is-section--md is-fluid grid hero">
        <div className="row">
          <div className="is-xs-12 column has-text-center">
            <PageHeader className="h3">{pkg.description}</PageHeader>
          </div>

          <div className="is-xs-12 column has-text-center">
            <Link to={downloadPath} className="button has-feather">
              Download <ChevronRight size={20} role="presentation" focusable="false" />
            </Link>
            <Link to={introductionPath} className="button is-primary has-gradient has-feather">
              Learn More <ChevronRight size={20} role="presentation" focusable="false" />
            </Link>
          </div>

          <div className="is-xs-12 column has-text-center">
            <p className="un-version has-no-m">Version {pkg.version}</p>
          </div>

          <div className="is-xs-12 column badges">
            <StatusBadges />
          </div>
        </div>
      </div>

      <div className="is-section--md is-fluid grid animations">
        <div className="row has-no-p">
          <div className="column has-no-p">
            <div className="wide grid">
              <ul className="row has-unstyled-list has-no-p">{renderAnimations()}</ul>
            </div>
          </div>
        </div>
      </div>

      <div className="is-section--md is-narrow grid">
        <div className="row">
          <div className="is-xs-12 column has-text-center">
            <h2 className="h6">Painless Setup</h2>
            <p>Install with npm:</p>
            <InstallNpm />
          </div>
          <div className="is-xs-12 column has-text-center">
            <p>Or simply link to minified assets:</p>
            <InstallAssets />
          </div>
          <div className="is-xs-12 column has-text-center">
            <p>See how Undernet can improve your developer experience!</p>
            <Link to={introductionPath} className="is-primary button has-gradient has-feather">
              Learn More <ChevronRight size={20} role="presentation" focusable="false" />
            </Link>
          </div>
        </div>
      </div>
    </div>
  )
}
