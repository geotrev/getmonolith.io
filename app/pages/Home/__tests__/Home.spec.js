import React from "react"
import { BrowserRouter as Router } from "react-router-dom"
import { LastLocationProvider } from "react-router-last-location"
import Home from "../Home"

jest.mock("app/components/ScrollUpOnMount", () => global.simpleMock("ScrollUpOnMount"))
jest.mock("app/components/SetMeta", () => global.simpleMock("SetMeta"))
jest.mock("app/components/PageHeader", () => global.simpleMock("PageHeader"))
jest.mock("react-feather/dist/icons/chevron-right", () => global.simpleMock("ChevronRight"))

jest.mock("../markdownContent", () => ({
  StatusBadges: global.simpleMock("StatusBadges"),
  InstallNpm: global.simpleMock("InstallNpm"),
  InstallAssets: global.simpleMock("Install"),
}))

jest.mock("projectRoot/package.json", () => ({
  description: "Mocked description",
  version: "9.9.9",
}))

jest.mock("prismjs", () => ({
  highlightAll: jest.fn(),
}))

jest.mock("lottie-web", () => ({
  loadAnimation: jest.fn().mockImplementation(() => ({
    play: jest.fn(),
    destroy: jest.fn(),
  })),
}))

const mountComponent = () => {
  return mount(
    <Router>
      <LastLocationProvider>
        <Home />
      </LastLocationProvider>
    </Router>
  )
}

describe("<Home />", () => {
  it("renders", () => {
    const wrapper = mountComponent()
    expect(wrapper).toMatchSnapshot()
  })
})
