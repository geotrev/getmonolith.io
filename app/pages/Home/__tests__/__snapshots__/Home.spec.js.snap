// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(LastLocationProvider)>
      <LastLocationProvider
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
        watchOnlyPathname={false}
      >
        <Home>
          <div
            id="home"
          >
            <ScrollUpOnMount />
            <SetMeta
              description="Undernet is a highly customizable web framework for building websites."
              title="A modular, configuration-first front-end framework. No strings."
            />
            <div
              className="medium-section fluid grid hero"
            >
              <div
                className="row"
              >
                <div
                  className="is-xsmall-12 column has-text-center"
                >
                  <PageHeader
                    className="h3"
                  >
                    Mocked description
                  </PageHeader>
                </div>
                <div
                  className="is-xsmall-12 column has-text-center"
                >
                  <Link
                    className="medium button has-feather"
                    to="/docs/overview/download"
                  >
                    <a
                      className="medium button has-feather"
                      href="/docs/overview/download"
                      onClick={[Function]}
                    >
                      Download 
                      <ChevronRight
                        focusable="false"
                        role="presentation"
                        size={20}
                      />
                    </a>
                  </Link>
                  <Link
                    className="primary medium button has-gradient has-feather"
                    to="/docs/overview/introduction"
                  >
                    <a
                      className="primary medium button has-gradient has-feather"
                      href="/docs/overview/introduction"
                      onClick={[Function]}
                    >
                      Learn More 
                      <ChevronRight
                        focusable="false"
                        role="presentation"
                        size={20}
                      />
                    </a>
                  </Link>
                </div>
                <div
                  className="is-xsmall-12 column has-text-center"
                >
                  <p
                    className="un-version has-no-margin"
                  >
                    Version 
                    9.9.9
                  </p>
                </div>
                <div
                  className="is-xsmall-12 column badges"
                >
                  <StatusBadges />
                </div>
              </div>
            </div>
            <div
              className="medium-section fluid grid animations"
            >
              <div
                className="row has-no-padding"
              >
                <div
                  className="column has-no-padding"
                >
                  <div
                    className="wide grid"
                  >
                    <ul
                      className="row has-unstyled-list has-no-padding"
                    >
                      <li
                        className="is-large-3 is-small-6 is-xsmall-12 column has-text-center has-no-padding-block-end"
                        key="Tiny"
                      >
                        <div
                          className="animated-icon"
                          id="animated-tiny"
                        />
                        <h2
                          className="h6 has-white-text"
                        >
                          Tiny
                        </h2>
                        <p
                          className="has-white-text"
                        >
                          CSS and JS under 14kb minified + gzipped; you can be assured performance isnâ€™t an issue.
                        </p>
                      </li>
                      <li
                        className="is-large-3 is-small-6 is-xsmall-12 column has-text-center has-no-padding-block-end"
                        key="Modular"
                      >
                        <div
                          className="animated-icon"
                          id="animated-modular"
                        />
                        <h2
                          className="h6 has-white-text"
                        >
                          Modular
                        </h2>
                        <p
                          className="has-white-text"
                        >
                          Remove and namespace pieces of the framework to prevent collisions and take only what you need.
                        </p>
                      </li>
                      <li
                        className="is-large-3 is-small-6 is-xsmall-12 column has-text-center has-no-padding-block-end"
                        key="Configurable"
                      >
                        <div
                          className="animated-icon"
                          id="animated-configurable"
                        />
                        <h2
                          className="h6 has-white-text"
                        >
                          Configurable
                        </h2>
                        <p
                          className="has-white-text"
                        >
                          Built for a great developer experience, you can customize and extend the library with ease.
                        </p>
                      </li>
                      <li
                        className="is-large-3 is-small-6 is-xsmall-12 column has-text-center has-no-padding-block-end"
                        key="Accessible"
                      >
                        <div
                          className="animated-icon"
                          id="animated-accessible"
                        />
                        <h2
                          className="h6 has-white-text"
                        >
                          Accessible
                        </h2>
                        <p
                          className="has-white-text"
                        >
                          Interactive components are designed with WAI-ARIA guidelines in mind to ensure your HTML is accessible.
                        </p>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="medium-section narrow grid"
            >
              <div
                className="row"
              >
                <div
                  className="is-xsmall-12 column has-text-center"
                >
                  <h2
                    className="h6"
                  >
                    Painless Setup
                  </h2>
                  <p>
                    Install with npm:
                  </p>
                  <InstallNpm />
                </div>
                <div
                  className="is-xsmall-12 column has-text-center"
                >
                  <p>
                    Or simply link to minified assets:
                  </p>
                  <Install />
                </div>
                <div
                  className="is-xsmall-12 column has-text-center"
                >
                  <p>
                    See how Undernet can improve your developer experience!
                  </p>
                  <Link
                    className="primary medium button has-gradient has-feather"
                    to="/docs/overview/introduction"
                  >
                    <a
                      className="primary medium button has-gradient has-feather"
                      href="/docs/overview/introduction"
                      onClick={[Function]}
                    >
                      Learn More 
                      <ChevronRight
                        focusable="false"
                        role="presentation"
                        size={20}
                      />
                    </a>
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </Home>
      </LastLocationProvider>
    </withRouter(LastLocationProvider)>
  </Router>
</BrowserRouter>
`;
